dm/syncer/syncer.go

```diff
@@ -2029,7 +2029,9 @@ func (s *Syncer) handleRowsEvent(ev *replication.RowsEvent, ec eventContext) err
 		return err2
 	}
 
-	prunedColumns, prunedRows, err := pruneGeneratedColumnDML(tableInfo, rows)
+	extRows := genrateExtendColumn(rows, s.tableRouter, sourceTable, s.cfg.SourceID)
```

Comment:
ditto, always `FetchExtendColumn` is not good for perf

Reply:
Hi @Ehco1996 @lance6716 , why is this conversation marked as resolved?

About route:
  The logic of [router.FetchExtendColumn](https://github.com/pingcap/tidb-tools/blob/cef6f500d02daca6f6e420e1e6d3466dfb39ddbd/pkg/table-router/router.go?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L263) and [router.Route](https://github.com/pingcap/tidb-tools/blob/cef6f500d02daca6f6e420e1e6d3466dfb39ddbd/pkg/table-router/router.go?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L185) is very similar, but more return fields are required. `router.Route` has already been called once in phases [syncer](https://github.com/pingcap/ticdc/blob/e924c71914bf04505a2cd0f002d95d95059eff1f/dm/syncer/syncer.go#L2995) and [loader](https://github.com/pingcap/ticdc/blob/e924c71914bf04505a2cd0f002d95d95059eff1f/dm/loader/loader.go?_pjax=%23js-repo-pjax-container%2C%20div%5Bitemtype%3D%22http%3A%2F%2Fschema.org%2FSoftwareSourceCode%22%5D%20main%2C%20%5Bdata-pjax-container%5D#L1234). For performance reasons, I can add the extended column return value of `router.Route` or make a new func return `targetSchema,targetTable,extendCols,extendVals` so that no routing operations are added. If this is necessary.

About cache:
Adding a cache for A is a good idea, but this will result in the cached data not being updated in time after the upstream table migrates the instance, which may result in the downstream extended column data being wrong. If this situation is tolerable, I think adding caching would be very helpful for performance.
One solution is to restart the dm task after the upstream table migration instance?

Reply:
i think it's better to optimise performance in next or another PR, i am afraid that this pr will get so large that it will be difficult to merge, so i marked this conversation as resolved

abourt cache or func change, IMO, maybe we need some perf profiling to prove thoses  optimzations are really useful